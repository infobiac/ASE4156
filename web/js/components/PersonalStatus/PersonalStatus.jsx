// @flow
import React from 'react';
import PropTypes from 'prop-types';
import Grid from 'material-ui/Grid';
import HighlightBox from '../HighlightBox';

import type { PersonalStatusRelay_bank }
  from './__generated__/PersonalStatusRelay_bank.graphql';
import type { PersonalStatusRelay_account } from './__generated__/PersonalStatusRelay_account.graphql';

type Props = {
  bank: PersonalStatusRelay_bank,
  account: PersonalStatusRelay_account
}

const spacing = { xs: 12, sm: 6, md: 3 };

export default class PersonalStatus extends React.Component < Props > {
  static propTypes = {
    bank: PropTypes.shape({
      balance: PropTypes.number.isRequired,
      income: PropTypes.number.isRequired,
      outcome: PropTypes.number.isRequired,
      monthlyStart: PropTypes.string.isRequired,
      monthlyEnd: PropTypes.string.isRequired,
      balanceDate: PropTypes.string.isRequired,
    }).isRequired,
  }
  render() {
    const secondary = [{ text: 'Start', value: this.props.bank.monthlyStart }, { text: 'End', value: this.props.bank.monthlyEnd }];
    return (
      <Grid container spacing={16} align="stretch" id="personal-status">
        <Grid item {...spacing}>
          <HighlightBox
            title={`Current balance (as of: ${this.props.bank.balanceDate})`}
            value={this.props.bank.balance.toFixed(2)}
            secondaryInfo={[{
              text: this.props.account.accountName,
              value: this.props.account.totalValue.toFixed(2),
            }]}
          />
        </Grid>
        <Grid item {...spacing}>
          <HighlightBox title="Monthly Income" value={this.props.bank.income.toFixed(2)} secondaryInfo={secondary} />
        </Grid>
        <Grid item {...spacing}>
          <HighlightBox title="Total Expenditures" value={this.props.bank.outcome.toFixed(2)} secondaryInfo={secondary} />
        </Grid>
        <Grid item {...spacing}>
          <HighlightBox
            title={`Available money (as of: ${this.props.bank.balanceDate})`}
            value={(this.props.bank.income + this.props.bank.outcome).toFixed(2)}
            secondaryInfo={[{
              text: this.props.account.accountName,
              value: this.props.account.availableCash.toFixed(2),
            }]}
          />
        </Grid>
      </Grid>
    );
  }
}
