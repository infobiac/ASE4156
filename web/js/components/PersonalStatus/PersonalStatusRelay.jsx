// @flow
import { createFragmentContainer, graphql } from 'react-relay';
import PersonalStatus from './PersonalStatus';

export default createFragmentContainer(PersonalStatus, {
  bank: graphql`
    fragment PersonalStatusRelay_bank on GUserBank {
      balance
      balanceDate
      income
      outcome
      monthlyStart
      monthlyEnd
    }
`,
  account: graphql`
    fragment PersonalStatusRelay_account on GTradingAccount {
      totalValue
      accountName
      availableCash
    }
  `,
});
