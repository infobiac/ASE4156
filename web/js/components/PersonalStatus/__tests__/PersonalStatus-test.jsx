import React from 'react';
import { shallow } from 'enzyme';
import PersonalStatus from '../PersonalStatus';

describe('PersonalStatus', () => {
  it('exists', () => {
    expect(PersonalStatus).toBeTruthy();
  });
  it('renders', () => {
    const comp = shallow((
      <PersonalStatus
        bank={{
 balance: 1.0, income: 2.0, outcome: 4.0, monthlyStart: '2016-01-01', monthlyEnd: '2016-02-01', balanceDate: '2016-02-01',
}}
        account={{ availableCash: 200.0, totalValue: 100.0, accountName: 'test' }}
      />
    ));
    expect(comp).toMatchSnapshot();
  });
});
