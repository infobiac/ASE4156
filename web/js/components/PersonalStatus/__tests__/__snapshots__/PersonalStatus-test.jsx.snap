// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalStatus renders 1`] = `
<withStyles(GridWrapper)
  align="stretch"
  container={true}
  id="personal-status"
  spacing={16}
>
  <withStyles(GridWrapper)
    item={true}
    md={3}
    sm={6}
    xs={12}
  >
    <HighlightBox
      secondaryInfo={
        Array [
          Object {
            "text": "test",
            "value": "100.00",
          },
        ]
      }
      title="Current balance (as of: 2016-02-01)"
      value="1.00"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    md={3}
    sm={6}
    xs={12}
  >
    <HighlightBox
      secondaryInfo={
        Array [
          Object {
            "text": "Start",
            "value": "2016-01-01",
          },
          Object {
            "text": "End",
            "value": "2016-02-01",
          },
        ]
      }
      title="Monthly Income"
      value="2.00"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    md={3}
    sm={6}
    xs={12}
  >
    <HighlightBox
      secondaryInfo={
        Array [
          Object {
            "text": "Start",
            "value": "2016-01-01",
          },
          Object {
            "text": "End",
            "value": "2016-02-01",
          },
        ]
      }
      title="Total Expenditures"
      value="4.00"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    md={3}
    sm={6}
    xs={12}
  >
    <HighlightBox
      secondaryInfo={
        Array [
          Object {
            "text": "test",
            "value": "200.00",
          },
        ]
      }
      title="Available money (as of: 2016-02-01)"
      value="6.00"
    />
  </withStyles(GridWrapper)>
</withStyles(GridWrapper)>
`;
